From c7fa3995cbaebc64ded5a8e9e04648fc6b6d8dcf Mon Sep 17 00:00:00 2001
From: Valerio Setti <vsetti@baylibre.com>
Date: Tue, 6 Jun 2023 17:57:04 +0200
Subject: [PATCH 38/41] irq: adding support for ethernet interrupts

Interrupt numbers were taken from MCAL code and checked against
the excel file with the full list of interrupts, so numbers should
be quite reliable.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>
---
 .../config/cortex-a/armv8/spider/config.oil          | 10 ++++++++++
 .../cortex-a/armv8/spider/interruptSources.oil       | 12 +++++++++++-
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/goil/templates/config/cortex-a/armv8/spider/config.oil b/goil/templates/config/cortex-a/armv8/spider/config.oil
index 918e7477..910f904f 100755
--- a/goil/templates/config/cortex-a/armv8/spider/config.oil
+++ b/goil/templates/config/cortex-a/armv8/spider/config.oil
@@ -34,6 +34,16 @@ CPU spider {
   // TO BE UPDATED
   INTERRUPT_SOURCE ARM_TIMER             { SOURCE_NUM = 27; VECTOR = IRQ_Interrupt; ACK = TRUE; };
   INTERRUPT_SOURCE ICCOM_INT             { SOURCE_NUM = 353; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA0_RX_TX_INT       { SOURCE_NUM = 312; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA1_RX_TX_INT       { SOURCE_NUM = 320; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA0_RX_TS_INT       { SOURCE_NUM = 328; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA1_RX_TS_INT       { SOURCE_NUM = 330; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE COMA_ERR_INT          { SOURCE_NUM = 290; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA0_ERR_INT         { SOURCE_NUM = 291; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE GWCA1_ERR_INT         { SOURCE_NUM = 292; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE ETHA0_ERR_INT         { SOURCE_NUM = 293; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE ETHA1_ERR_INT         { SOURCE_NUM = 294; VECTOR = IRQ_Interrupt; ACK = TRUE; };
+  INTERRUPT_SOURCE ETHA2_ERR_INT         { SOURCE_NUM = 295; VECTOR = IRQ_Interrupt; ACK = TRUE; };
 
   PLATFORM_FILES spider {
     PATH = "cortex-a/armv8/spider";
diff --git a/goil/templates/config/cortex-a/armv8/spider/interruptSources.oil b/goil/templates/config/cortex-a/armv8/spider/interruptSources.oil
index 07aef502..cb298815 100755
--- a/goil/templates/config/cortex-a/armv8/spider/interruptSources.oil
+++ b/goil/templates/config/cortex-a/armv8/spider/interruptSources.oil
@@ -4,5 +4,15 @@
 ENUM [
   /* ARM IRQs */
   ARM_TIMER,              /* 0 */
-  ICCOM_INT               /* 1 */
+  ICCOM_INT,              /* 1 */
+  GWCA0_RX_TX_INT,        /* 2 */
+  GWCA1_RX_TX_INT,        /* 3 */
+  GWCA0_RX_TS_INT,        /* 4 */
+  GWCA1_RX_TS_INT,        /* 5 */
+  COMA_ERR_INT,           /* 6 */
+  GWCA0_ERR_INT,          /* 7 */
+  GWCA1_ERR_INT,          /* 8 */
+  ETHA0_ERR_INT,          /* 9 */
+  ETHA1_ERR_INT,          /* 10 */
+  ETHA2_ERR_INT           /* 11 */
 ] SOURCE;
-- 
2.25.1

